### DISCUSSION TITLE
[Bug]  io.grpc.internal.AbstractServerImplBuilder Missing, Causing Server-Side Trace Failure

### DISCUSSION THREAD

**Discussion by cdming405003** [apache/skywalking#12585] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

Hello,

I've encountered an issue when upgrading gRPC to a version higher than [1.58.0]. It seems that the io.grpc.internal.AbstractServerImplBuilder class no longer exists in these versions. Skywalking's AbstractServerImplBuilderInstrumentation only enhances io.grpc.internal.AbstractServerImplBuilder.

### What you expected to happen

The server-side tracing functionality is now broken.

### How to reproduce

Upgrading gRPC to a version higher than [1.58.0]

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
We only verified 1.25 since the plugin contributed.

Check here, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/grpc-scenario/support-version.list

So, if you face issues at 1.58(far after 1.25), it is surprising. You may need to find the issue and submit a patch to upstream. Or you have to wait for someone picks this and fixes it.

**Comment by cdming405003** [comment]
My current solution is to enhance the **build** method of **io.grpc.ForwardingServerBuilder**. But the final solution should be to enhance the **abstract build** method of **io.grpc.ServerBuilder** class. I tried it but failed. I would like to ask if skywalking supports enhancing abstract methods. I hope you can provide some ideas. @wu-sheng

**Reply by wu-sheng** [reply]
There is no abstract method enhancement. Because this method needs the implementations to enhance.

You could enhance the method in all the implementations, but not abstract method itself.

**Reply by cdming405003** [reply]
Got it, thanks!
